src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,387 @@
+        /// <returns>Dictionary with property name and property value in the skiptoken value.</returns>
+        protected static IDictionary<string, object> PopulatePropertyValuePairs(string value, ODataQueryContext context)
+        {
+            Contract.Assert(context != null);
+
+            IDictionary<string, object> propertyValuePairs = new Dictionary<string, object>();
+            string[] keyValues = value.Split(CommaDelimiter);
+            foreach (string keyAndValue in keyValues)
+            {
+                string[] pieces = keyAndValue.Split(new char[] { propertyDelimiter }, 2);
+                if (pieces.Length > 1 && !String.IsNullOrWhiteSpace(pieces[1]))
+                {
+                    object propValue = null;
+                    if (pieces[1].StartsWith("'enumType'"))
+                    {
+                        string enumValue = pieces[1].Remove(0, 10);
+                        IEdmTypeReference type = EdmLibHelpers.GetTypeReferenceOfProperty(context.Model, context.ElementClrType, pieces[0]);
+                        propValue = ODataUriUtils.ConvertFromUriLiteral(enumValue, ODataVersion.V401, context.Model, type);
+                    }
+                    else
+                    {
+                        propValue = ODataUriUtils.ConvertFromUriLiteral(pieces[1], ODataVersion.V401);
+                    }
+                    if (!String.IsNullOrWhiteSpace(pieces[0]))
+                    {
+                        propertyValuePairs.Add(pieces[0], propValue);
+                    }
+                }
+            }
+
+            return propertyValuePairs;
+        }
+
+        /// <summary>
+        /// Returns the list of properties that should be used for generating the skiptoken value. 
+        /// </summary>
+        /// <param name="lastMember">The last record that will be returned in the response.</param>
+        /// <param name="model">IEdmModel</param>
+        /// <param name="orderByNodes">OrderBy nodes in the original request.</param>
+        /// <returns>List of properties that should be used for generating the skiptoken value.</returns>
+        protected static IEnumerable<IEdmProperty> GetPropertiesForSkipToken(object lastMember, IEdmModel model, IList<OrderByNode> orderByNodes)
+        {
+            IEdmType edmType = GetTypeFromObject(lastMember, model);
+            IEdmEntityType entity = edmType as IEdmEntityType;
+            if (entity == null)
+            {
+                return null;
+            }
+
+            IList<IEdmProperty> key = entity.Key().AsIList<IEdmProperty>();
+            if (orderByNodes != null)
+            {
+                OrderByOpenPropertyNode orderByOpenType = orderByNodes.OfType<OrderByOpenPropertyNode>().LastOrDefault();
+                if (orderByOpenType != null)
+                {
+                    //SkipToken will not support ordering on dynamic properties
+                    return null;
+                }
```

Comment:
if just test the orderByNodes contains "dynamic properties", why can't use "Any(..)"